### PR TITLE
LT-2521: Add currencies

### PR DESCRIPTION
**Pull request by gponomarev-lykke**
Based on https://github.com/lykkecloud/MarginTrading.AssetService/pull/27

### CODE REVIEW

**Comment by gponomarev-lykke** [comment]
Had to rebase on https://github.com/lykkecloud/MarginTrading.AssetService/pull/33 to implement CurrencyChangedEvent

File: src/MarginTrading.AssetService.Core/Services/ICurrenciesService.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using MarginTrading.AssetService.Core.Domain;
+
+namespace MarginTrading.AssetService.Core.Services
+{
+    public interface ICurrenciesService
+    {
+        Task<Result<CurrenciesErrorCodes>> InsertAsync(Currency currency, string username, string correlationId);
+        Task<Result<CurrenciesErrorCodes>> UpdateAsync(Currency currency, string username, string correlationId);
+        Task<Result<CurrenciesErrorCodes>> DeleteAsync(string id, string username, string correlationId);
+        Task<Result<Currency, CurrenciesErrorCodes>> GetByIdAsync(string mdsCode);
```
Comment (andrey-anokhin) [inline src/MarginTrading.AssetService.Core/Services/ICurrenciesService.cs:13]: Why it is called mdsCore?

File: src/MarginTrading.AssetService.Core/Services/ICurrenciesService.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        Task<Result<List<Currency>, CurrenciesErrorCodes>> GetByPageAsync(int skip = default, int take = 20);
+        Task<Result<List<Currency>, CurrenciesErrorCodes>> GetAllAsync();
```
Comment (andrey-anokhin) [inline src/MarginTrading.AssetService.Core/Services/ICurrenciesService.cs:15]: Did not get the idea of why we have 2 different methods, but with the same response types?

File: src/MarginTrading.AssetService.Core/Services/ICurrenciesService.cs (hunk 1 above)
Comment (gponomarev-lykke) [reply src/MarginTrading.AssetService.Core/Services/ICurrenciesService.cs:13]: Will fix

File: src/MarginTrading.AssetService.Core/Services/ICurrenciesService.cs (hunk 2 above)
Comment (gponomarev-lykke) [reply src/MarginTrading.AssetService.Core/Services/ICurrenciesService.cs:15]: The first one gets paginated result, the second one gets all the records.
I moved the code responsible for choosing the correct method to the controller, looks more logical there.

**Review by andrey-anokhin — APPROVED** [review]

**Pull request merged**